ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -170,6 +170,23 @@ func (r *RayClusterReconciler) Reconcile(ctx context.Context, request ctrl.Reque
 	return ctrl.Result{}, client.IgnoreNotFound(err)
 }
 
+func (r *RayClusterReconciler) DeleteAllPods(ctx context.Context, namespace string, filterLabels client.MatchingLabels) (active int, err error) {
+	pods := corev1.PodList{}
+	if err = r.List(ctx, &pods, client.InNamespace(namespace), filterLabels); err != nil {
+		return 0, err
+	}
+	active = 0
+	for _, pod := range pods.Items {
+		if pod.DeletionTimestamp == nil || pod.DeletionTimestamp.IsZero() {
```

Comment:
https://github.com/kubernetes/apimachinery/blob/46a872ec555cd60a5bc172ce28d9ae0234266aae/pkg/apis/meta/v1/time.go#L58-L64

`pod.DeletionTimestamp == nil` seems to be redundant.

Reply:
Oh, that's handy!